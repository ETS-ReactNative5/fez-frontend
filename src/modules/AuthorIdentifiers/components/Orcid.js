import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Cookies from 'js-cookie';
import {parse, stringify} from 'querystring';
import {createHash} from 'crypto';

import {locale, routes} from 'config';
import {APP_URL, SESSION_COOKIE_NAME, ORCID_CLIENT_ID, ORCID_AUTHORIZATION_URL} from 'config/general';

import RaisedButton from 'material-ui/RaisedButton';
import {StandardPage, StandardCard, ConfirmDialogBox, Alert} from 'uqlibrary-react-toolbox';

export default class Orcid extends Component {
    static propTypes = {
        account: PropTypes.object,
        locale: PropTypes.object,
        history: PropTypes.object,
        location: PropTypes.object,
        actions: PropTypes.object
    };

    constructor(props) {
        super(props);
        this.state = {
            error: null
        };
    }

    componentDidMount() {
        const {orcid} = locale.authorIdentifiers;

        const queryParams = parse(window.location.search.substr(1));

        if (queryParams.code && queryParams.state && queryParams.state !== this.getState()) {
            this.setError(orcid.stateErrorAlert);
        } else if (queryParams.code) {
            this.props.actions.requestAuthorOrcidInfo(this.props.account.id, {code: queryParams.code, redirUri: this.props.location.pathname});
            this.props.history.push(routes.pathConfig.dashboard);
        }
    }

    _setAuthoriseConfirmation = (ref) => {
        this.authoriseConfirmationBox = ref;
    };

    _showAuthoriseConfirmation = (orcidUrlGenerator) => {
        const url = orcidUrlGenerator();
        this.authoriseConfirmationBox._onAction = () => (window.location.assign(url));
        this.authoriseConfirmationBox.showConfirmation();
    };

    _authoriseOrcid = () => {
        return this.requestPermissionUrl(this.props.location.pathname, {show_login: true});
    };

    _createOrcid = () => {
        const additionalParams = {
            show_login: false,
            family_names: this.props.account.lastName,
            given_names: this.props.account.firstName
        };
        return this.requestPermissionUrl(this.props.location.pathname, additionalParams);
    };

    setError = (error) => {
        this.setState({error: error});
    };

    getState = () => (createHash('md5').update(Cookies.get(SESSION_COOKIE_NAME)).digest('hex'));

    requestPermissionUrl = (returnUrl, additionalParams = {}) => {
        const params = {
            client_id: ORCID_CLIENT_ID,
            response_type: 'code',
            redirect_uri: `${APP_URL}#${returnUrl}`,
            state: this.getState(),
            scope: '/authenticate',
            ...additionalParams
        };

        return `${ORCID_AUTHORIZATION_URL}?${stringify(params)}`;
    };

    render() {
        const txt = locale.authorIdentifiers.orcid;

        return (
            <StandardPage title={txt.title}>
                <form onKeyDown={this._handleKeyboardFormSubmit}>
                    <ConfirmDialogBox
                        onRef={this._setAuthoriseConfirmation}
                        locale={txt.grantAccessConfirmation} />
                    <StandardCard title={txt.authoriseOrcid.title}>
                        <div>{txt.authoriseOrcid.description}</div>
                        <div className="columns action-buttons">
                            <div className="column is-hidden-mobile"/>
                            <div className="column is-narrow-desktop">
                                <RaisedButton
                                    secondary
                                    fullWidth
                                    label={txt.authoriseOrcid.labels.submit}
                                    onTouchTap={() => this._showAuthoriseConfirmation(this._authoriseOrcid)}
                                />
                            </div>
                        </div>
                    </StandardCard>
                    <StandardCard title={txt.createOrcid.title}>
                        <div>{txt.createOrcid.description}</div>
                        <div className="columns action-buttons">
                            <div className="column is-hidden-mobile"/>
                            <div className="column is-narrow-desktop">
                                <RaisedButton
                                    secondary
                                    fullWidth
                                    label={txt.createOrcid.labels.submit}
                                    onTouchTap={() => this._showAuthoriseConfirmation(this._createOrcid)}
                                />
                            </div>
                        </div>
                    </StandardCard>

                    {
                        this.state.error &&
                        <Alert {...this.state.error} />
                    }
                </form>
            </StandardPage>
        );
    }
}
