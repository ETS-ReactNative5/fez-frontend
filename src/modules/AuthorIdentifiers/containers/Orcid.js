import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { withRouter } from 'react-router-dom';
import * as actions from 'actions';

import Orcid from '../components/Orcid';

const mapStateToProps = state => ({
    ...state.get('accountReducer'),
});

const mapDispatchToProps = dispatch => ({
    actions: bindActionCreators(actions, dispatch),
});

let OrcidContainer = connect(mapStateToProps, mapDispatchToProps)(Orcid);
OrcidContainer = withRouter(OrcidContainer);

export default OrcidContainer;
