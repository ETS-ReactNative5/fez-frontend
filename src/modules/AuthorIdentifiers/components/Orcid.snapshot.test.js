jest.mock('js-cookie', () => {
    function get(something) {
        return something || 'test'
    }

    return {
        get: get
    }
});

jest.dontMock('./Orcid');

import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Orcid from './Orcid';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import PropTypes from 'prop-types';
import injectTapEventPlugin from 'react-tap-event-plugin';

function setup({author, history, actions, location, account, isShallow = false}) {
    const props = {
        author: author || {aut_id: 410},
        actions: actions || {},
        history: history || {},
        location: location || {},
        account: account || {firstName: 'Test', lastName: 'Last'}
    };

    if(isShallow) {
        return shallow(<Orcid {...props} />);
    }

    return mount(<Orcid {...props} />, {
        context: {
            muiTheme: getMuiTheme()
        },
        childContextTypes: {
            muiTheme: PropTypes.object.isRequired
        }
    });

}

beforeAll(() => {
    injectTapEventPlugin();
});


describe('Component Orcid ', () => {
    it('should render claim publication form', () => {
        const wrapper = setup({});
        expect(wrapper.find('RaisedButton').length).toEqual(2);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should open confirmation dialog box for link orcid', () => {
        const wrapper = setup({history: {}});
        const showConfirmation = jest.fn();
        wrapper.instance().authoriseConfirmationBox = {
            showConfirmation: showConfirmation
        };

        wrapper.instance()._confirmLinkOrcid();
        expect(showConfirmation).toBeCalled();
    });

    it('should open confirmation dialog box for create orcid', () => {
        const wrapper = setup({history: {}});
        const showConfirmation = jest.fn();
        wrapper.instance().authoriseConfirmationBox = {
            showConfirmation: showConfirmation
        };

        wrapper.instance()._confirmCreateOrcid();
        expect(showConfirmation).toBeCalled();
    });
});
