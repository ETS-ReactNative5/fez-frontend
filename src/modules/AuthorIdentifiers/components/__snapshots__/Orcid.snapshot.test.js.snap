// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Orcid  should render claim publication form 1`] = `
<Orcid
  account={
    Object {
      "firstName": "Test",
      "lastName": "Last",
    }
  }
  actions={Object {}}
  author={
    Object {
      "aut_id": 410,
    }
  }
  history={Object {}}
  location={Object {}}
>
  <StandardPage
    title="Link ORCID ID to UQ eSpace"
  >
    <div
      className="undefined"
    >
      <h1
        className="pageTitle"
      >
        Link ORCID ID to UQ eSpace
      </h1>
      <div
        className="layout-card"
      >
        <ConfirmDialogBox
          hideCancelButton={false}
          locale={
            Object {
              "cancelButtonLabel": "Cancel",
              "confirmButtonLabel": "OK",
              "confirmationMessage": <div>
                <p>
                  In order to proceed you will now be directed to ORCID.org website.
                </p>
                <p>
                  You will be redirected back after you have granted UQ access.
                </p>
              </div>,
              "confirmationTitle": "ORCID Grant Access",
            }
          }
          onRef={[Function]}
        >
          <Dialog
            actions={
              Array [
                <div
                  className="columns dialog-actions ConfirmDialogBox-actions"
                >
                  <div
                    className="column is-hidden-mobile"
                  />
                  <div
                    className="column is-narrow"
                  >
                    <RaisedButton
                      className="ConfirmDialogBox-actions-cancel"
                      disabled={false}
                      fullWidth={true}
                      label="Cancel"
                      labelPosition="after"
                      onTouchTap={[Function]}
                      primary={false}
                      secondary={false}
                    />
                  </div>
                  <div
                    className="column is-narrow"
                  >
                    <RaisedButton
                      className="ConfirmDialogBox-actions-confirm"
                      disabled={false}
                      fullWidth={true}
                      keyboardFocused={true}
                      label="OK"
                      labelPosition="after"
                      onTouchTap={[Function]}
                      primary={false}
                      secondary={true}
                    />
                  </div>
                  <div
                    className="is-clearfix"
                  />
                </div>,
              ]
            }
            autoDetectWindowHeight={true}
            autoScrollBodyContent={false}
            modal={true}
            open={false}
            repositionOnUpdate={true}
            title="ORCID Grant Access"
          >
            <RenderToLayer
              open={true}
              render={[Function]}
              useLayerForClickAway={false}
            />
          </Dialog>
        </ConfirmDialogBox>
        <StandardCard
          className=""
          title="I already have an ORCID iD"
        >
          <Card
            className=" standard-card"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
          >
            <Paper
              circle={false}
              className=" standard-card"
              rounded={true}
              style={
                Object {
                  "zIndex": 1,
                }
              }
              transitionEnabled={true}
              zDepth={1}
            >
              <div
                className=" standard-card"
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "backgroundColor": "#ffffff",
                    "borderRadius": 2,
                    "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "Roboto, sans-serif",
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "zIndex": 1,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "paddingBottom": 8,
                    }
                  }
                >
                  <CardHeader
                    avatar={null}
                    className="card-header"
                  >
                    <div
                      className="card-header"
                      style={
                        Object {
                          "boxSizing": "border-box",
                          "fontWeight": 500,
                          "muiPrepared": true,
                          "padding": 16,
                          "position": "relative",
                          "whiteSpace": "nowrap",
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "muiPrepared": true,
                            "paddingRight": "90px",
                            "verticalAlign": "top",
                            "whiteSpace": "normal",
                          }
                        }
                      >
                        <span
                          style={
                            Object {
                              "color": "rgba(0, 0, 0, 0.87)",
                              "display": "block",
                              "fontSize": 15,
                              "muiPrepared": true,
                            }
                          }
                        />
                        <span
                          style={
                            Object {
                              "color": "rgba(0, 0, 0, 0.54)",
                              "display": "block",
                              "fontSize": 14,
                              "muiPrepared": true,
                            }
                          }
                        />
                      </div>
                      <div
                        className="columns is-gapless is-mobile"
                      >
                        <div
                          className="column"
                        >
                          <h2
                            className="cardTitle"
                          >
                            I already have an ORCID iD
                          </h2>
                        </div>
                      </div>
                    </div>
                  </CardHeader>
                  <CardText
                    className="body-1"
                  >
                    <div
                      className="body-1"
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontSize": 14,
                          "muiPrepared": true,
                          "padding": 16,
                        }
                      }
                    >
                      <div>
                        <div>
                          <div>
                            This option enables you to link your existing ORCID iD to UQ.
                          </div>
                        </div>
                        <div
                          className="columns action-buttons"
                        >
                          <div
                            className="column is-hidden-mobile"
                          />
                          <div
                            className="column is-narrow-desktop"
                          >
                            <RaisedButton
                              disabled={false}
                              fullWidth={true}
                              label="Link your ORCID iD"
                              labelPosition="after"
                              onTouchTap={[Function]}
                              primary={false}
                              secondary={true}
                            >
                              <Paper
                                circle={false}
                                rounded={true}
                                style={
                                  Object {
                                    "display": "inline-block",
                                    "minWidth": "100%",
                                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                  }
                                }
                                transitionEnabled={true}
                                zDepth={1}
                              >
                                <div
                                  style={
                                    Object {
                                      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                                      "backgroundColor": "#ffffff",
                                      "borderRadius": 2,
                                      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                                      "boxSizing": "border-box",
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "display": "inline-block",
                                      "fontFamily": "Roboto, sans-serif",
                                      "minWidth": "100%",
                                      "muiPrepared": true,
                                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                    }
                                  }
                                >
                                  <EnhancedButton
                                    containerElement="button"
                                    disabled={false}
                                    focusRippleColor="#ffffff"
                                    focusRippleOpacity={0.16}
                                    onBlur={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    onKeyUp={[Function]}
                                    onKeyboardFocus={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseEnter={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchStart={[Function]}
                                    onTouchTap={[Function]}
                                    style={
                                      Object {
                                        "backgroundColor": "#ff4081",
                                        "borderRadius": 2,
                                        "height": 36,
                                        "lineHeight": "36px",
                                        "padding": 0,
                                        "textAlign": "center",
                                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                        "width": "100%",
                                      }
                                    }
                                    tabIndex={0}
                                    touchRippleColor="#ffffff"
                                    touchRippleOpacity={0.16}
                                    type="button"
                                  >
                                    <button
                                      disabled={false}
                                      onBlur={[Function]}
                                      onClick={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchStart={[Function]}
                                      onTouchTap={[Function]}
                                      style={
                                        Object {
                                          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                                          "backgroundColor": "#ff4081",
                                          "border": 10,
                                          "borderRadius": 2,
                                          "boxSizing": "border-box",
                                          "cursor": "pointer",
                                          "display": "inline-block",
                                          "fontFamily": "Roboto, sans-serif",
                                          "fontSize": "inherit",
                                          "fontWeight": "inherit",
                                          "height": 36,
                                          "lineHeight": "36px",
                                          "margin": 0,
                                          "muiPrepared": true,
                                          "outline": "none",
                                          "padding": 0,
                                          "position": "relative",
                                          "textAlign": "center",
                                          "textDecoration": "none",
                                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                          "verticalAlign": null,
                                          "width": "100%",
                                          "zIndex": 1,
                                        }
                                      }
                                      tabIndex={0}
                                      type="button"
                                    >
                                      <TouchRipple
                                        abortOnScroll={true}
                                        color="#ffffff"
                                        opacity={0.16}
                                      >
                                        <div
                                          onMouseDown={[Function]}
                                          onMouseLeave={[Function]}
                                          onMouseUp={[Function]}
                                          onTouchEnd={[Function]}
                                          onTouchStart={[Function]}
                                        >
                                          <div
                                            style={
                                              Object {
                                                "backgroundColor": false,
                                                "borderRadius": 2,
                                                "height": 36,
                                                "muiPrepared": true,
                                                "top": 0,
                                                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                              }
                                            }
                                          >
                                            <span
                                              style={
                                                Object {
                                                  "color": "#ffffff",
                                                  "fontSize": 14,
                                                  "fontWeight": 500,
                                                  "letterSpacing": 0,
                                                  "margin": 0,
                                                  "muiPrepared": true,
                                                  "opacity": 1,
                                                  "paddingLeft": 16,
                                                  "paddingRight": 16,
                                                  "position": "relative",
                                                  "textTransform": "uppercase",
                                                  "userSelect": "none",
                                                }
                                              }
                                            >
                                              Link your ORCID iD
                                            </span>
                                          </div>
                                        </div>
                                      </TouchRipple>
                                    </button>
                                  </EnhancedButton>
                                </div>
                              </Paper>
                            </RaisedButton>
                          </div>
                        </div>
                      </div>
                    </div>
                  </CardText>
                </div>
              </div>
            </Paper>
          </Card>
        </StandardCard>
        <StandardCard
          className=""
          title="I need an ORCID iD"
        >
          <Card
            className=" standard-card"
            expandable={false}
            expanded={null}
            initiallyExpanded={false}
          >
            <Paper
              circle={false}
              className=" standard-card"
              rounded={true}
              style={
                Object {
                  "zIndex": 1,
                }
              }
              transitionEnabled={true}
              zDepth={1}
            >
              <div
                className=" standard-card"
                style={
                  Object {
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "backgroundColor": "#ffffff",
                    "borderRadius": 2,
                    "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                    "boxSizing": "border-box",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "fontFamily": "Roboto, sans-serif",
                    "muiPrepared": true,
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                    "zIndex": 1,
                  }
                }
              >
                <div
                  style={
                    Object {
                      "paddingBottom": 8,
                    }
                  }
                >
                  <CardHeader
                    avatar={null}
                    className="card-header"
                  >
                    <div
                      className="card-header"
                      style={
                        Object {
                          "boxSizing": "border-box",
                          "fontWeight": 500,
                          "muiPrepared": true,
                          "padding": 16,
                          "position": "relative",
                          "whiteSpace": "nowrap",
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "display": "inline-block",
                            "muiPrepared": true,
                            "paddingRight": "90px",
                            "verticalAlign": "top",
                            "whiteSpace": "normal",
                          }
                        }
                      >
                        <span
                          style={
                            Object {
                              "color": "rgba(0, 0, 0, 0.87)",
                              "display": "block",
                              "fontSize": 15,
                              "muiPrepared": true,
                            }
                          }
                        />
                        <span
                          style={
                            Object {
                              "color": "rgba(0, 0, 0, 0.54)",
                              "display": "block",
                              "fontSize": 14,
                              "muiPrepared": true,
                            }
                          }
                        />
                      </div>
                      <div
                        className="columns is-gapless is-mobile"
                      >
                        <div
                          className="column"
                        >
                          <h2
                            className="cardTitle"
                          >
                            I need an ORCID iD
                          </h2>
                        </div>
                      </div>
                    </div>
                  </CardHeader>
                  <CardText
                    className="body-1"
                  >
                    <div
                      className="body-1"
                      style={
                        Object {
                          "color": "rgba(0, 0, 0, 0.87)",
                          "fontSize": 14,
                          "muiPrepared": true,
                          "padding": 16,
                        }
                      }
                    >
                      <div>
                        <div>
                          <div>
                            <p>
                              This option enables you to create a new ORCID iD and link it with UQ.
                            </p>
                            <p>
                              Use this option if you are unsure if you already have an ORCID iD. It will detect matches to your name and email from the ORCID registry and prompt you to log in to avoid creating a new ORCID iD.
                            </p>
                          </div>
                        </div>
                        <div
                          className="columns action-buttons"
                        >
                          <div
                            className="column is-hidden-mobile"
                          />
                          <div
                            className="column is-narrow-desktop"
                          >
                            <RaisedButton
                              disabled={false}
                              fullWidth={true}
                              label="Create a new ORCID iD"
                              labelPosition="after"
                              onTouchTap={[Function]}
                              primary={false}
                              secondary={true}
                            >
                              <Paper
                                circle={false}
                                rounded={true}
                                style={
                                  Object {
                                    "display": "inline-block",
                                    "minWidth": "100%",
                                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                  }
                                }
                                transitionEnabled={true}
                                zDepth={1}
                              >
                                <div
                                  style={
                                    Object {
                                      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                                      "backgroundColor": "#ffffff",
                                      "borderRadius": 2,
                                      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
                                      "boxSizing": "border-box",
                                      "color": "rgba(0, 0, 0, 0.87)",
                                      "display": "inline-block",
                                      "fontFamily": "Roboto, sans-serif",
                                      "minWidth": "100%",
                                      "muiPrepared": true,
                                      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                    }
                                  }
                                >
                                  <EnhancedButton
                                    containerElement="button"
                                    disabled={false}
                                    focusRippleColor="#ffffff"
                                    focusRippleOpacity={0.16}
                                    onBlur={[Function]}
                                    onClick={[Function]}
                                    onFocus={[Function]}
                                    onKeyDown={[Function]}
                                    onKeyUp={[Function]}
                                    onKeyboardFocus={[Function]}
                                    onMouseDown={[Function]}
                                    onMouseEnter={[Function]}
                                    onMouseLeave={[Function]}
                                    onMouseUp={[Function]}
                                    onTouchEnd={[Function]}
                                    onTouchStart={[Function]}
                                    onTouchTap={[Function]}
                                    style={
                                      Object {
                                        "backgroundColor": "#ff4081",
                                        "borderRadius": 2,
                                        "height": 36,
                                        "lineHeight": "36px",
                                        "padding": 0,
                                        "textAlign": "center",
                                        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                        "width": "100%",
                                      }
                                    }
                                    tabIndex={0}
                                    touchRippleColor="#ffffff"
                                    touchRippleOpacity={0.16}
                                    type="button"
                                  >
                                    <button
                                      disabled={false}
                                      onBlur={[Function]}
                                      onClick={[Function]}
                                      onFocus={[Function]}
                                      onKeyDown={[Function]}
                                      onKeyUp={[Function]}
                                      onMouseDown={[Function]}
                                      onMouseEnter={[Function]}
                                      onMouseLeave={[Function]}
                                      onMouseUp={[Function]}
                                      onTouchEnd={[Function]}
                                      onTouchStart={[Function]}
                                      onTouchTap={[Function]}
                                      style={
                                        Object {
                                          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
                                          "backgroundColor": "#ff4081",
                                          "border": 10,
                                          "borderRadius": 2,
                                          "boxSizing": "border-box",
                                          "cursor": "pointer",
                                          "display": "inline-block",
                                          "fontFamily": "Roboto, sans-serif",
                                          "fontSize": "inherit",
                                          "fontWeight": "inherit",
                                          "height": 36,
                                          "lineHeight": "36px",
                                          "margin": 0,
                                          "muiPrepared": true,
                                          "outline": "none",
                                          "padding": 0,
                                          "position": "relative",
                                          "textAlign": "center",
                                          "textDecoration": "none",
                                          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                          "verticalAlign": null,
                                          "width": "100%",
                                          "zIndex": 1,
                                        }
                                      }
                                      tabIndex={0}
                                      type="button"
                                    >
                                      <TouchRipple
                                        abortOnScroll={true}
                                        color="#ffffff"
                                        opacity={0.16}
                                      >
                                        <div
                                          onMouseDown={[Function]}
                                          onMouseLeave={[Function]}
                                          onMouseUp={[Function]}
                                          onTouchEnd={[Function]}
                                          onTouchStart={[Function]}
                                        >
                                          <div
                                            style={
                                              Object {
                                                "backgroundColor": false,
                                                "borderRadius": 2,
                                                "height": 36,
                                                "muiPrepared": true,
                                                "top": 0,
                                                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                                              }
                                            }
                                          >
                                            <span
                                              style={
                                                Object {
                                                  "color": "#ffffff",
                                                  "fontSize": 14,
                                                  "fontWeight": 500,
                                                  "letterSpacing": 0,
                                                  "margin": 0,
                                                  "muiPrepared": true,
                                                  "opacity": 1,
                                                  "paddingLeft": 16,
                                                  "paddingRight": 16,
                                                  "position": "relative",
                                                  "textTransform": "uppercase",
                                                  "userSelect": "none",
                                                }
                                              }
                                            >
                                              Create a new ORCID iD
                                            </span>
                                          </div>
                                        </div>
                                      </TouchRipple>
                                    </button>
                                  </EnhancedButton>
                                </div>
                              </Paper>
                            </RaisedButton>
                          </div>
                        </div>
                      </div>
                    </div>
                  </CardText>
                </div>
              </div>
            </Paper>
          </Card>
        </StandardCard>
      </div>
    </div>
  </StandardPage>
</Orcid>
`;
